medusa/process_tv.py

```diff
@@ -432,14 +430,20 @@ def unrar(self, path, rar_files, force=False):
                 self._log('Unpacking archive: {0}'.format(archive), logger.DEBUG)
 
                 failure = None
+                # Use custom unrar path only if set, else will use from $PATH
+                if app.UNRAR_TOOL:
+                    rarfile.UNRAR_TOOL = app.UNRAR_TOOL
                 try:
-                    rar_handle = RarFile(os.path.join(path, archive))
+                    rar_handle = rarfile.RarFile(os.path.join(path, archive))
+
+                    # Test rar, if fails it will raise an exception
+                    rar_handle.testrar()
 
                     # Skip extraction if any file in archive has previously been extracted
                     skip_extraction = False
                     for file_in_archive in [os.path.basename(each.filename)
                                             for each in rar_handle.infolist()
```

Comment:
namelist calls infolist. Why do you **prefer** namelist?

If you don't have the patience to do a proper review, how do you expect other devs to join and code? As almost no one else is coding/improving here except @p0psicles 

rarfile.py
```python
    def namelist(self):
        """Return list of filenames in archive.
        """
        return [f.filename for f in self.infolist()]

    def infolist(self):
        """Return RarInfo objects for all files/directories in archive.
        """
        return self._file_parser.infolist()
```